File: kotlin/src/test/kotlin/pages/Homepage.kt
package pages

import com.microsoft.playwright.Locator.GetByRoleOptions
import com.microsoft.playwright.Page
import com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat
import com.microsoft.playwright.options.AriaRole.*
import java.util.regex.Pattern
import java.util.regex.Pattern.CASE_INSENSITIVE

class Homepage(private val page: Page) {
    fun `accept cookies`(): Homepage {
        page.getByRole(BUTTON, Page.GetByRoleOptions().setName("Accept")).click()
        return this
    }

    fun `close privacy policy`(): Homepage {
        val policyIframe = page.frameLocator("iframe[title='Popup CTA']")
        policyIframe.locator("css=.button-container").click()
        return this
    }

    fun `pick SecureExchange solution`(): SecureExchange {
        val nav = page.getByRole(NAVIGATION)
        nav.getByRole(BUTTON, GetByRoleOptions().setName("Solutions")).hover()
        nav.getByRole(LINK, GetByRoleOptions().setName("SecureExchange")).click()
        assertThat(page).hasTitle("SecureExchange")
        return SecureExchange(page)
    }

    companion object {
        fun navigate(page: Page): Homepage {
            page.navigate("https://dswiss.com/en")
            assertThat(page).hasTitle(
                Pattern.compile(
                    "DSwiss – We make your business processes simple and secure",
                    CASE_INSENSITIVE
                )
            );
            return Homepage(page)
        }
    }
}

File: typescript/tests/pages/homepage.ts
import { expect, Page } from "@playwright/test"
import { SecureExchange } from "./secureExchange"
import { step } from "../testStep"

export class Homepage {
  constructor(private page: Page) {}

  @step
  static async navigate(page: Page) {
    await page.goto("/")
    await expect(page).toHaveTitle(/DSwiss – We make your business processes simple and secure/i)
    return new Homepage(page)
  }

  @step
  async acceptCookies() {
    await this.page.getByRole("button", { name: "Accept" }).click()
  }

  @step
  async closePrivacyPolicy() {
    const policyIframe = this.page.frameLocator("iframe[title='Popup CTA']")
    await policyIframe.locator("css=.button-container").click()
  }

  @step
  async pickSecureExchangeSolution() {
    const nav = this.page.getByRole("navigation")
    await nav.getByRole("button", { name: "Solutions" }).hover()
    await nav.getByRole("link", { name: "SecureExchange" }).click()
    await expect(this.page).toHaveTitle("SecureExchange")
    return new SecureExchange(this.page)
  }
}
